src/Symfony/Bundle/FrameworkBundle/Command/UidGenerateCommand.php

```diff
@@ -0,0 +1,139 @@
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> generates UID. This can be a ULID or a UUID
+in a given version. Available types are $typesAsString.
+Examples:
+
+  <info>php %command.full_name% ulid</info> for generating a ULID.
+  <info>php %command.full_name% 1</info> for generating a UUID in version 1.
+  <info>php %command.full_name% 3 9b7541de-6f87-11ea-ab3c-9da9a81562fc foo</info> for generating a UUID in version 3.
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \LogicException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if (!class_exists(AbstractUid::class)) {
+            throw new \RuntimeException('Unable to execute this command as the Symfony Uid Component is not installed.');
+        }
+
+        $type = $input->getArgument('type') ? strtolower($input->getArgument('type')) : null;
+        $namespace = $input->getArgument('namespace');
+        $name = $input->getArgument('name');
+
+        if (null === $type || !\in_array($type, self::$types, true)) {
+            $type = $io->ask('Type/version of the UID. Available values are '.implode(', ', self::$types).'.', null, function ($type) {
+                $type = strtolower($type);
+                if (!\in_array($type, self::$types, true)) {
+                    throw new \RuntimeException('Available values are '.implode(', ', self::$types).'.');
+                }
+
+                return $type;
+            });
+        }
+
+        if (\in_array($type, [self::UUID_3, self::UUID_5], true) && (null === $namespace || !Uuid::isValid($namespace))) {
+            $namespace = $io->ask('Please enter a valid namespace:', null, function ($namespace) {
+                if (null === $namespace || !Uuid::isValid($namespace)) {
+                    throw new \RuntimeException('This is not a valid namespace.');
+                }
+
+                return $namespace;
+            });
+        }
+
+        if (\in_array($type, [self::UUID_3, self::UUID_5], true) && empty($name)) {
+            $name = $io->ask('Please enter a name. Press Enter for an empty string. ', '');
+        }
+
+        switch ($type) {
+            case self::UUID_1:
+                $uid = Uuid::v1();
+                break;
+            case self::UUID_3:
+                $uid = Uuid::v3(Uuid::fromString($namespace), $name);
+                break;
+            case self::UUID_4:
+                $uid = Uuid::v4();
+                break;
+            case self::UUID_5:
+                $uid = Uuid::v5(Uuid::fromString($namespace), $name);
+                break;
+            case self::UUID_6:
+                $uid = Uuid::v6();
+                break;
+            case self::ULID:
+                $uid = new Ulid();
+                break;
+        }
+
+        $io->title('Generated UID:');
```

Comment:
Just a thought: I understand that you want to provide a friendly user interface. However, this command could be consumed by shell scripts if we provided a way to produce a UUID without any additional interaction or output.

Reply:
Maybe we should (if that possible) output everything except from the UID to stderr instead of stdout?

Reply:
I've updated the code, so it just outputs the result now.